corehq/apps/reports/tests/test_case_list_report.py

```diff
@@ -0,0 +1,142 @@
+            dummy_case_list,
+            dummy_user_list,
+        )
+
+        for user in dummy_user_list:
+            user_obj = CouchUser.get_by_username(user['username'])
+            if user_obj:
+                user_obj.delete('')
+        cls.user_list = []
+        for user in dummy_user_list:
+            user_obj = CommCareUser.create(**user) if user['doc_type'] == 'CommcareUser'\
+                else WebUser.create(**user)
+            user_obj.save()
+            cls.user_list.append(user_obj)
+
+        cls.case_list = []
+        for case in dummy_case_list:
+            cls.case_list.append(CommCareCase(**case))
+        cls.es = get_es_new()
+        ensure_index_deleted(USER_INDEX)
+        ensure_index_deleted(CASE_INDEX)
+        initialize_index_and_mapping(cls.es, USER_INDEX_INFO)
+        initialize_index_and_mapping(cls.es, CASE_INDEX_INFO)
+        initialize_index_and_mapping(cls.es, GROUP_INDEX_INFO)
+        cls._send_users_to_es()
+        cls._send_cases_to_es()
+
+    def setUp(self):
+        super().setUp()
+        self.request = self.request_factory.get('/some/url')
+        self.request.couch_user = self.user
+        self.request.domain = self.domain
+        self.request.can_access_all_locations = True
+
+    @classmethod
+    def tearDownClass(cls):
+        ensure_index_deleted(USER_INDEX)
+        ensure_index_deleted(CASE_INDEX)
+        for user in cls.user_list:
+            user.delete(deleted_by='')
+        super().tearDownClass()
+
+    @classmethod
+    def _send_users_to_es(cls):
+        for user_obj in cls.user_list:
+            send_to_elasticsearch('users', transform_user_for_elasticsearch(user_obj.to_json()))
+        cls.es.indices.refresh(USER_INDEX)
+
+    @classmethod
+    def _send_cases_to_es(cls):
+        for case in cls.case_list:
+            send_to_elasticsearch('cases', case.to_json())
+        cls.es.indices.refresh(CASE_INDEX_INFO.index)
+
+    def test_with_project_data_slug(self):
+        report_slugs = ['project_data']
+        q_dict_get = QueryDict('', mutable=True)
+        q_dict_get.setlist('case_list_filter', report_slugs)
+        self.request.GET = q_dict_get
+        data = CaseListReport(self.request, domain=self.domain).es_results['hits'].get('hits', [])
+        expected_case_ids = ['id-1', 'id-2', 'id-3', 'id-5']
+        queried_case_ids = [case['_id'] for case in data]
+        sorted(queried_case_ids)
+        self.assertCountEqual(expected_case_ids, queried_case_ids)
+
+    def test_with_deactivated_slug(self):
+        report_slugs = ['t__5']
+        q_dict_get = QueryDict('', mutable=True)
+        q_dict_get.setlist('case_list_filter', report_slugs)
+        self.request.GET = q_dict_get
+        data = CaseListReport(self.request, domain=self.domain).es_results['hits'].get('hits', [])
+        expected_case_ids = ['id-1']
+        queried_case_ids = [case['_id'] for case in data]
+        sorted(queried_case_ids)
```

Comment:
out of curiosity, why do you sort these? I think you can use `assertItemsEqual` to check that two lists have the same things in them.

Reply:
I guess just for sanity 😅  
`assertItemEqual` has been replaced by `assertCountEqual` in Python 3, I have used that here and I agree removing sorted will be a good idea here.
Thanks for pointing it out.

Reply:
Whoa, that's a terrible name for that method 😄 
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

>Test that sequence first contains the same elements as second, regardless of their order.

Reply:
Done

Reply:
> Whoa, that's a terrible name for that method 😄

Yeah. I was confused at first and thought it has something to do with the count of elements.